File: Python/quicksort.py
def partition (a, s, e):  
    i = (s - 1)  
    pivot = a[e] 
      
    for j in range(s, e):  
     
        if (a[j] <= pivot):  
            i = i + 1  
            a[i], a[j] = a[j], a[i]  
      
    a[i+1], a[e] = a[e], a[i+1]  
  
    return (i + 1)  

def quick(a, s, end): 
    if (s < end):  
        p = partition(a, s, end) 
        quick(a, s, p - 1)  
        quick(a, p + 1, end)  
  
          
def printArr(a): 
    for i in range(len(a)):  
        print (a[i], end = " ")  
  
      
a = [68, 1003, 10, 9, 508, 121,524,138,687]  
print("Before sorting array elements are - ")  
printArr(a)  
quick(a, 0, len(a)-1)  
print("\nAfter sorting array elements are - ")  
printArr(a)  

File: CPP/QuickSort.cpp
#include <iostream>

namespace sorting
{
    int partition(int arr[], int low, int high)
    {
        int pivot = arr[high];
        int i = (low - 1);

        for (int j = low; j < high; j++)
        {
            if (arr[j] <= pivot)
            {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
        int temp = arr[i + 1];
        arr[i + 1] = arr[high];
        arr[high] = temp;
        return (i + 1);
    }
    void quickSort(int arr[], int low, int high)
    {
        if (low < high)
        {
            int p = partition(arr, low, high);
            quickSort(arr, low, p - 1);
            quickSort(arr, p + 1, high);
        }
    }

}

int main()
{
    int arr[] = {324, 45, 34, 54645675, 8678675, 9, 869, 879, 78, 978, 59, 56, 73, 54, 25, 43, 432151, 64575};
    int size = sizeof(arr) / sizeof(arr[0]);
    sorting::quickSort(arr, 0, size);
    for (int i = 0; i < size; i++)
    {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;
    return 0;
}